engine/servermaster/service_util.go

```diff
@@ -56,3 +57,41 @@ func generateNodeID(name string) string {
 	id := fmt.Sprintf("%s-%08x", name, val)
 	return id
 }
+
```

Comment:
```suggestion
var _ rpcutil.FeatureChecker = &featureDegrader{}
```
Besides, perhaps leaderFeatureChecker would be a better name?

Reply:
I use `featureDegrader` here because we may add more downgrade policy in server master, `rpcutil.FeatureChecker` is part of them.